azurerm/internal/services/containers/kubernetes_cluster_data_source.go

```diff
@@ -938,6 +940,10 @@ func flattenKubernetesClusterDataSourceAgentPoolProfiles(input *[]containerservi
 			agentPoolProfile["tags"] = tags.Flatten(profile.Tags)
 		}
 
+		if profile.UpgradeSettings != nil {
+			agentPoolProfile["upgrade_settings"] = flattenUpgradeSettings(profile.UpgradeSettings)
+		}
+
 		agentPoolProfiles = append(agentPoolProfiles, agentPoolProfile)
```

Comment:
this is a larger issue with this data source, but these fields need to be set in all circumstances - I'll push a commit to fix this